### PR TITLE
virtctl,version: Remove globals

### PR DESCRIPTION
**Pull request by oshoval**
### What this PR does
Embed the globals as part of the version struct,
in order to get rid from globals, which is a better practice.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by oshoval** [comment]
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/12913/pull-kubevirt-unit-test/1838537289433616384

is not related, reported about it (fails due to the order of the expected elements)

File: pkg/virtctl/version/version.go [hunk 1]
```diff
@@ -15,25 +15,25 @@ import (
 	"kubevirt.io/kubevirt/pkg/virtctl/templates"
 )
 
-var (
-	cmd        *cobra.Command
-	clientOnly bool
-)
+type Version struct {
```
Comment (0xFelix) [inline pkg/virtctl/version/version.go:18]: You could make this not exported.

Comment (oshoval) [reply pkg/virtctl/version/version.go]: done

**Comment by oshoval** [comment]
@akalenyu 
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/12913/pull-kubevirt-e2e-k8s-1.31-sig-storage/1838545829783146496

reminds what we had with SR-IOV isnt it ?

**Comment by akalenyu** [comment]
> Hi @akalenyu https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/12913/pull-kubevirt-e2e-k8s-1.31-sig-storage/1838545829783146496
> 
> reminds what we had with SR-IOV isnt it ?

Yeah looks like it, I can see it for other lanes too
https://search.ci.kubevirt.io/?search=SyncVMI%3A+rpc+error%3A+code+%3D+DeadlineExceeded+desc+%3D+context+deadline+exceeded&maxAge=336h&context=1&type=bug%2Bissue%2Bjunit&name=&excludeName=storage&maxMatches=5&maxBytes=20971520&groupBy=job
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/12817/pull-kubevirt-e2e-k8s-1.30-sig-network/1835316483828224000

**Comment by oshoval** [comment]
> > Hi @akalenyu https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/12913/pull-kubevirt-e2e-k8s-1.31-sig-storage/1838545829783146496
> > reminds what we had with SR-IOV isnt it ?
> 
> Yeah looks like it, I can see it for other lanes too https://search.ci.kubevirt.io/?search=SyncVMI%3A+rpc+error%3A+code+%3D+DeadlineExceeded+desc+%3D+context+deadline+exceeded&maxAge=336h&context=1&type=bug%2Bissue%2Bjunit&name=&excludeName=storage&maxMatches=5&maxBytes=20971520&groupBy=job https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/12817/pull-kubevirt-e2e-k8s-1.30-sig-network/1835316483828224000

@EdDev 
FYI, it affects also sig-network, maybe worth to check if same solution of the SR-IOV case can be applied please

**Comment by EdDev** [comment]
@akalenyu , @oshoval , can you please be explicit here?

I digged into one failure and saw this:
`[FAILED] Unexpected Warning event received: testvmi-t29xr,e56a3596-04c1-4e85-b87b-fed669e1c002: unknown error encountered sending command SyncVMI: rpc error: code = DeadlineExceeded desc = context deadline exceeded`

So you refer to the problem seen on the SR-IOV lane regarding slow response from the cluster and our nasty workaround to ignore it. Is that it?

If this spreads all over, then we have a problem and I prefer not to ignore it.
But if everyone feels comfortable with ignoring these all over the e2e tests, lets discuss it in the sig-ci context.

**Comment by oshoval** [comment]
> @akalenyu , @oshoval , can you please be explicit here?
> 
> I digged into one failure and saw this: `[FAILED] Unexpected Warning event received: testvmi-t29xr,e56a3596-04c1-4e85-b87b-fed669e1c002: unknown error encountered sending command SyncVMI: rpc error: code = DeadlineExceeded desc = context deadline exceeded`
> 
> So you refer to the problem seen on the SR-IOV lane regarding slow response from the cluster and our nasty workaround to ignore it. Is that it?
> 
> If this spreads all over, then we have a problem and I prefer not to ignore it. But if everyone feels comfortable with ignoring these all over the e2e tests, lets discuss it in the sig-ci context.

Ye it just reminds what we had on SR-IOV, so a spike to find out the exact flow might worth it.
I don't know if non SR-IOV also waits for that file that caused the race (it isn't i believe, else we would see the new logs that were added).

**Comment by oshoval** [comment]
@jcanocan 
can you please take a look ?

**Comment by jcanocan** [comment]
> @jcanocan can you please take a look ?

Thanks for cleaning efforts.
Please, add some commit message, PR description and release note.
Apart from that, it looks good to me.

**Comment by oshoval** [comment]
> > @jcanocan can you please take a look ?
> 
> Thanks for cleaning efforts. Please, add some commit message, PR description and release note. Apart from that, it looks good to me.

Done

**Comment by jcanocan** [comment]
Thanks!
/lgtm

**Review by 0xFelix — APPROVED** [review]
/approve

**Pull request merged**